features/step_definitions/shf_documents_steps.rb

```diff
@@ -0,0 +1,20 @@
+When(/^I choose a shf-document named "([^"]*)" to upload$/) do | filename |
+  page.attach_file "shf_document[actual_file]", File.join(Rails.root, 'spec', 'fixtures','uploaded_files', filename)
+end
+
+
+And(/^I should see (\d+) shf-documents listed$/) do |number|
+  expect(page).to have_selector('.shf-document', count: number)
```

Comment:
I have started to prefer Capybara's `assert` methods over Rspec `expect`.  This is because the `assert` methods will wait for the configured capybara wait time - for an entity to show up on the page - before giving up and raising an exception.  

Spec (as far as I know) does not respect Capybara's wait time.

http://www.rubydoc.info/github/jnicklas/capybara/master/Capybara/Node/Matchers#assert_selector-instance_method